server/src/main/java/org/tctalent/server/api/portal/CandidateExamPortalApi.java

```diff
@@ -0,0 +1,69 @@
+package org.tctalent.server.api.portal;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.tctalent.server.api.admin.ITableApi;
+import org.tctalent.server.exception.EntityExistsException;
+import org.tctalent.server.exception.InvalidRequestException;
+import org.tctalent.server.exception.NoSuchObjectException;
+import org.tctalent.server.model.db.CandidateExam;
+import org.tctalent.server.request.candidate.exam.CreateCandidateExamRequest;
+import org.tctalent.server.request.candidate.exam.UpdateCandidateExamsRequest;
+import org.tctalent.server.service.db.CandidateExamService;
+import org.tctalent.server.util.dto.DtoBuilder;
+
+@RestController
+@RequestMapping("/api/portal/candidate-exam")
+public class CandidateExamPortalApi implements ITableApi<CreateCandidateExamRequest,UpdateCandidateExamsRequest,UpdateCandidateExamsRequest> {
```

Comment:
You have not implemented the correct interface - because CandidateExam is joined to Candidate, the correct interface to implement is IJoinedTableApi - which supports the concept of "parent ids". The parent id is the id of the parent Candidate record.